src/CMakeLists.txt

```diff
@@ -0,0 +1,323 @@
+    ${CMAKE_CURRENT_SOURCE_DIR}/zxing/zxing/qrcode/decoder/DataMask.h
+    ${CMAKE_CURRENT_SOURCE_DIR}/zxing/zxing/qrcode/decoder/DecodedBitStreamParser.h
+    ${CMAKE_CURRENT_SOURCE_DIR}/zxing/zxing/qrcode/decoder/Decoder.h
+    ${CMAKE_CURRENT_SOURCE_DIR}/zxing/zxing/qrcode/decoder/Mode.h
+    ${CMAKE_CURRENT_SOURCE_DIR}/zxing/zxing/qrcode/decoder/QRBitMatrixParser.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/zxing/zxing/qrcode/decoder/QRDataBlock.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/zxing/zxing/qrcode/decoder/QRDataMask.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/zxing/zxing/qrcode/decoder/QRDecodedBitStreamParser.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/zxing/zxing/qrcode/decoder/QRDecoder.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/zxing/zxing/qrcode/decoder/QRMode.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/zxing/zxing/qrcode/detector/AlignmentPattern.h
+    ${CMAKE_CURRENT_SOURCE_DIR}/zxing/zxing/qrcode/detector/AlignmentPatternFinder.h
+    ${CMAKE_CURRENT_SOURCE_DIR}/zxing/zxing/qrcode/detector/Detector.h
+    ${CMAKE_CURRENT_SOURCE_DIR}/zxing/zxing/qrcode/detector/FinderPattern.h
+    ${CMAKE_CURRENT_SOURCE_DIR}/zxing/zxing/qrcode/detector/FinderPatternFinder.h
+    ${CMAKE_CURRENT_SOURCE_DIR}/zxing/zxing/qrcode/detector/FinderPatternInfo.h
+    ${CMAKE_CURRENT_SOURCE_DIR}/zxing/zxing/qrcode/detector/QRAlignmentPattern.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/zxing/zxing/qrcode/detector/QRAlignmentPatternFinder.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/zxing/zxing/qrcode/detector/QRDetector.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/zxing/zxing/qrcode/detector/QRFinderPattern.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/zxing/zxing/qrcode/detector/QRFinderPatternFinder.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/zxing/zxing/qrcode/detector/QRFinderPatternInfo.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/zxing/zxing/qrcode/encoder/BlockPair.h
+    ${CMAKE_CURRENT_SOURCE_DIR}/zxing/zxing/qrcode/encoder/ByteMatrix.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/zxing/zxing/qrcode/encoder/ByteMatrix.h
+    ${CMAKE_CURRENT_SOURCE_DIR}/zxing/zxing/qrcode/encoder/Encoder.h
+    ${CMAKE_CURRENT_SOURCE_DIR}/zxing/zxing/qrcode/encoder/MaskUtil.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/zxing/zxing/qrcode/encoder/MaskUtil.h
+    ${CMAKE_CURRENT_SOURCE_DIR}/zxing/zxing/qrcode/encoder/MatrixUtil.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/zxing/zxing/qrcode/encoder/MatrixUtil.h
+    ${CMAKE_CURRENT_SOURCE_DIR}/zxing/zxing/qrcode/encoder/QRCode.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/zxing/zxing/qrcode/encoder/QRCode.h
+    ${CMAKE_CURRENT_SOURCE_DIR}/zxing/zxing/qrcode/encoder/QREncoder.cpp
+
+    )
+
+target_link_libraries(qzxing
+    Qt5::Core
+    Qt5::Widgets
+    Qt5::Svg
+    Qt5::Quick
+    Qt5::QuickControls2
+    Qt5::Multimedia
+    Qt5::MultimediaWidgets )
+
+file(GLOB PUBLIC_HEADER QZXing.h)
+
+# Change Global Definitions depending on how you want to use the library
+target_compile_definitions(qzxing
+  PRIVATE -QZXING_LIBRARY
+  PRIVATE -ZXING_ICONV_CONST
+  PUBLIC  -QZXING_QML
+  PUBLIC  -QZXING_MULTIMEDIA
```

Comment:
Same, Qml amd Multimedia should be configurable, enabled on demand..?

Reply:
I added new parameters:
-QZXING_MULTIMEDIA set ON for use  QZXingFilter
-QZXING_USE_QML set ON for use QZXING + QML

you can try now there: 8eefa1d653d29729a8336fa8d09284e23b50bf2e
When I do more settings from pri file, I'll send next pull request.